.github/workflows/e2e-test.yml

```diff
@@ -0,0 +1,46 @@
+name: Local Env CI
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  local-env:
+    name: Setup Local Environment
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout Local Env repository
+      uses: actions/checkout@v4
+
+    - name: Setup and Start Local Env Dev Container
+      uses: devcontainers/ci@v0.3
+      with:
+        runCmd: |
+          kind create cluster --name to-dos --config kind-local-config.yaml --kubeconfig ./.to-dos-cluster-kubeconfig
+          helmfile cache cleanup && helmfile --environment local --namespace local -f deploy/helmfile.yaml apply
+
+    - name: Checkout UI repository
+      uses: actions/checkout@v4
+      with:
+        repository: FaceMan01/to-dos-ui
+        path: to-dos-ui
+
+    - name: Setup and Start UI Dev Container
```

Comment:
You access to-dos cluster that was just created in another job's step. Recently we faced an issue that it is not guaranteed that what was created in one step will survive until all job's steps complete. Can you please check it? I'm not quite sure if it is indeed true and not guaranteed by GitHub Actions.

Reply:
As I understand it, within a single job, all changes persist from step to step. But, different jobs operate on separate systems, and changes are not survive from one job to another. I faced this when I tried to separate the creation of a Dev Container for the local env and a Dev Container for the UI into different jobs.